src/shared_realm.cpp

```diff
@@ -364,10 +364,13 @@ void Realm::update_schema(Schema schema, uint64_t version, MigrationFunction mig
     // Either the schema version has changed or we need to do non-migration changes
 
     m_group->set_schema_change_notification_handler(nullptr);
-    transaction::begin_without_validation(*m_shared_group);
+    if (!in_transaction) {
+        transaction::begin_without_validation(*m_shared_group);
+    }
     add_schema_change_handler();
 
     // Cancel the write transaction if we exit this function before committing it
+    // FIXME: don't cancel if in_transaction == true
```

Comment:
Why add this FIXME rather than adding the `!in_transaction` check to `WriteTransactionGuard`'s destructor? Won't the behavior as-is result in some subtle bugs?